### PR TITLE
Allow projects to opt-out of source build

### PR DESCRIPTION
**Pull request by tmat**
Implements `ExcludeFromSourceBuild` property that projects can set to exclude themselves from source build.

### CODE REVIEW

**Comment by tmat** [comment]
@nguerrera @dsplaisted @weshaggard @jaredpar This is slightly hacky but I think it's better than maintaining extra source-build-only solutions or adding conditions to each project that multi-targets. 

It would be nice if NuGet Restore had more supported way to skip restore of a project from within the project. I could only find `_IsProjectRestoreSupported` target that does what we need here.

File: sdks/RepoToolset/sdk/Sdk.targets [hunk 1]
```diff
@@ -4,6 +4,35 @@
...
+  -->
+
+  <PropertyGroup>
+    <_SuppressBuild>false</_SuppressBuild>
+    <_SuppressBuild Condition="'$(__ExcludeSdkImports)' == 'true'">true</_SuppressBuild>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(DotNetBuildFromSource)' == 'true' and !$(_SuppressBuild)">
+    <_ReducedTargetFrameworks Condition="'$(TargetFrameworks)' != ''">$([System.Text.RegularExpressions.Regex]::Replace('$(TargetFrameworks)', 'net[0-9]+;?', '').Trim(';'))</_ReducedTargetFrameworks>
+    <_ReducedTargetFrameworks Condition="'$(TargetFrameworks)' == ''">$([System.Text.RegularExpressions.Regex]::Replace('$(TargetFramework)', 'net[0-9]+', ''))</_ReducedTargetFrameworks>
```
Comment (jaredpar) [inline sdks/RepoToolset/sdk/Sdk.targets]: We want to keep all the net* targets though. RedHat was okay with us doing that as we're just compiling against the equivalent of header files here.

Comment (tmat) [reply sdks/RepoToolset/sdk/Sdk.targets]: I see. The projects that reference non-FX non-OSS assemblies need to be filtered out (anything that builds against VS assemblies, for example), right?

Comment (tmat) [reply sdks/RepoToolset/sdk/Sdk.targets]: Updated the filter to check for `ExcludeFromSourceBuild`. Projects can now set this property to exclude themselves from source build.

**Comment by tmat** [comment]
test ci please

**Comment by tmat** [comment]
test this please

**Pull request merged**